frontend/public/components/namespace.jsx

```diff
@@ -210,49 +209,64 @@ const namespaceDropdownStateToProps = state => {
   return { activeNamespace, canListNS };
 };
 
-const NamespaceDropdown = connect(namespaceDropdownStateToProps)(props => {
-  const { activeNamespace, dispatch, canListNS, useProjects } = props;
-  if (canListNS === undefined) {
-    return null;
-  }
 
-  const { loaded, data } = props.namespace;
-  const model = getModel(useProjects);
-  const allNamespacesTitle = `all ${model.labelPlural.toLowerCase()}`;
-  const items = {};
-  if (canListNS) {
-    items[ALL_NAMESPACES_KEY] = allNamespacesTitle;
+class NamespaceDropdown_ extends React.Component {
+
+  componentDidUpdate() {
+    const { namespace, dispatch } = this.props;
+    if (namespace.loaded) {
+      const projectsAvailable = !_.isEmpty(namespace.data);
+      setFlag(dispatch, FLAGS.PROJECTS_AVAILABLE, projectsAvailable);
+    }
   }
-  _.map(data, 'metadata.name').sort().forEach(name => items[name] = name);
-
-  let title = activeNamespace;
-  if (activeNamespace === ALL_NAMESPACES_KEY) {
-    title = allNamespacesTitle;
-  } else if (loaded && !_.has(items, title)) {
-    // If the currently active namespace is not found in the list of all namespaces, put it in anyway
-    items[title] = title;
+
+  render() {
+    const { activeNamespace, dispatch, canListNS, useProjects } = this.props;
+    if (canListNS === undefined) {
```

Comment:
Can `canListNS` be anything besides `undefined`?  `null`, or other falsy values?

Reply:
Yes, and they mean different things. We need to check specifically for `undefined `here.

@TheRealJon prefer `flagPending` from #208

Reply:
That was where I was going :)

Def prefer `flagPending()` now that we have it.